Windows: Compilation error on master branch

I'm trying to compile the master branch, at its most recent commit, of the Deal.II on a Windows machine with Visual Studio, but the compilation process dies due to the following error:

```
dealii\include\deal.ii\base\thread_management.h(267): error C3646: 'barrier': unknown override specifier.
dealii\include\deal.ii\base\thread_management.h(267): error C4430: missing type specifier - int assumed.
```

Here is the file reporting the build configuration: [detailed.log](https://github.com/dealii/dealii/files/2553445/detailed.log)

I investigated the problem a little bit, and this is what I found out so far:

The error seems to be triggered by an unrecognized data type, namely `pthread_barrier_t`. This is odd to me, since in Windows Visual Studio `<pthread.h>` doesn't exist, and I'm not sure it makes sense, for CMake, to generate a solution inclusive of pthread-dependent code.

Prior versions of Deal.II compile without problems, so I decided to compare the `thread_management.h` files from the *master branch* and *v.9.0.1*. 
In *v.9.0.1*, It look like that the line of code causing the error are inside a block `#ifdef DEAL_II_WITH_THREADS ... #endif`. While, in the *master branch* version, the line of code causing the error is not enclosed by a `#ifdef ... #endif` guarding block anymore.

--- a/include/deal.II/base/thread_management.h
+++ b/include/deal.II/base/thread_management.h
@@ -215,7 +215,7 @@ namespace Threads
     std::condition_variable condition_variable;
   };
 
-
+#  ifdef DEAL_II_USE_MT_POSIX
   /**
    * Implementation of a thread barrier class, based on the POSIX thread
    * functions. POSIX barriers are a relatively new feature and are not
@@ -263,11 +263,11 @@ namespace Threads
      * Data object storing the POSIX data which we need to call the POSIX
      * functions.
      */
-#  ifndef DEAL_II_USE_MT_POSIX_NO_BARRIERS
+#    ifndef DEAL_II_USE_MT_POSIX_NO_BARRIERS
     pthread_barrier_t barrier;
-#  else
+#    else
     unsigned int count;
-#  endif
+#    endif
   };
 
 
@@ -279,6 +279,7 @@ namespace Threads
    *   is deprecated.
    */
   using Barrier DEAL_II_DEPRECATED = PosixThreadBarrier;
+#  endif
 } // namespace Threads
 
 

